onnx/defs/controlflow/defs.cc

```diff
@@ -272,6 +272,107 @@ void IfInferenceFunction(InferenceContext& ctx) {
   }
 }
 
+void LoopInferenceFunction(InferenceContext& ctx) {
+  auto num_inputs = ctx.getNumInputs();
+  auto num_loop_state_vars = num_inputs - 2; // skip 'M' and 'cond'
+
+  std::vector<const TypeProto*> subgraph_input_types;
+  subgraph_input_types.push_back(nullptr); // iteration number. not inferred.
```

Comment:
Adding the expected numeric type would catch a type-error where the subgraph's first parameter is not the right type. (The spec doesn't clarify the type, but int64 seems best, and we could probably clarify the spec too.)

Reply:
Should that be handled here or in the checker where it validates types match the constraints?

Reply:
Sorry - I see what you mean now. The checker would only validate the max iterations type not this. Will add a check it's int64.